Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt

```diff
@@ -88,23 +88,27 @@ class ConfirmCheckInViewModel @AssistedInject constructor(
         ),
         completed: Boolean = false,
         createJournalEntry: Boolean = true
-    ): CheckIn = CheckIn(
-        traceLocationId = verifiedTraceLocation.traceLocationID,
-        traceLocationIdHash = verifiedTraceLocation.traceLocationID.sha256(),
-        version = traceLocation.version,
-        type = traceLocation.type.number,
-        description = traceLocation.description,
-        address = traceLocation.address,
-        traceLocationStart = traceLocation.startDate,
-        traceLocationEnd = traceLocation.endDate,
-        defaultCheckInLengthInMinutes = traceLocation.defaultCheckInLengthInMinutes,
-        cryptographicSeed = traceLocation.cryptographicSeed,
-        cnPublicKey = traceLocation.cnPublicKey,
-        checkInStart = checkInStart,
-        checkInEnd = checkInEnd,
-        completed = completed,
-        createJournalEntry = createJournalEntry,
-    )
+    ): CheckIn {
+        val traceLocationId = verifiedTraceLocation.traceLocation.locationId
+        return CheckIn(
+            traceLocationId = traceLocationId,
+            // sha256() Hash of locationId ,which is itself a sha256() Hash
+            traceLocationIdHash = traceLocationId.sha256(),
```

Comment:
Maybe also add the hashing to `TraceLocation` where we also have the calculation and hashing of the `traceLocationId`?

Reply:
I added what is common to the TraceLocation , locationIdHash is more for checkIn than Location , and it is accessed only once on  confirmation afaik, is it required somewhere else?

Reply:
`locationId` is also only needed for check-ins, or? I just think that the logic for `locationId` and `locationIdHash` should be in the same place.

Reply:
✅